python/ray/serve/_private/proxy_state.py

```diff
@@ -165,11 +181,13 @@ def _drain_check(self):
         assert self._status == ProxyStatus.DRAINING
 
         if self._is_drained_obj_ref:
-            finished, _ = ray.wait([self._is_drained_obj_ref], timeout=0)
+            finished, _ = self._ray_api_wrapper.wait(
+                [self._is_drained_obj_ref], timeout=0
+            )
```

Comment:
Instead of mocking the Ray API, here you could have a method on a proxy wrapper class such as: `is_drained()`. In testing you can directly implement this method without worrying about the Ray API at all.